packages/ozone-material/ozone-login-form/src/ozone-login-form.ts

```diff
@@ -0,0 +1,162 @@
+
+	@property({ type: String })
+	public messageOnEmpty: string = 'The username and/or password is empty'
+
+	@property({ type: String })
+	public messageOnInvalid: string = 'Invalid username and/or password'
+
+	@property({ type: String })
+	public messageOnUnknown: string = 'Unknown error. Please check your connection'
+
+	@property({ type: String, notify: true })
+	public alignButtons: string = 'center'
+
+	@property({ type: String })
+	public usernameLabel: string = 'Username'
+
+	@property({ type: String })
+	public passwordLabel: string = 'Password'
+
+	@property({ type: String })
+	public submitLabel: string = 'Login'
+
+	@property({ type: String })
+	public lostPasswordLabel: string = 'Lost password'
+
+	@property({ type: Boolean })
+	public showLostPassword: boolean = true
+
+	@property({ type: String })
+	public registerLabel: string = 'Register'
+
+	@property({ type: Boolean })
+	public showRegister: boolean = true
+	/**
+	 * Indicate if the user is connected.
+	 * This property can be watch.
+	 */
+	@property({ type: Boolean, notify: true, reflectToAttribute: true })
+	public isConnected: boolean = false
+
+	public register(e: Event): void {
+		this.dispatchEvent(new CustomEvent('register-clicked', {}))
+	}
+
+	public lostPassword(e: Event): void {
+		this.dispatchEvent(new CustomEvent('password-clicked', { detail: { username: this.username } }))
+	}
+	ready(): void {
+		super.ready()
+		const defaultClient = getDefaultClient()
+		defaultClient.onEnterState(ClientStates.STOPPED,() => {
+			this.set('isConnected', false)
+		})
+		defaultClient.onEnterState(ClientStates.AUTHENTICATED,() => {
+			this.set('isConnected', true)
+		})
+	}
+
+	public async submitForm(e: Event): Promise<void> {
+		if (this.dispatchEvent(new CustomEvent('before-submit', { detail: { form: this.$.form }, cancelable: true }))) {
+			this.validateForm()
+			const updateMessagesOnError = once(() => {
+				this.set('isConnected', false)
+				const err = defaultClient.lastFailedLogin
+				if (err && err.status === 400) {
+					this.set('errorMessage', this.messageOnEmpty)
+				} else if (err && err.status === 403) {
+					this.set('errorMessage', this.messageOnInvalid)
+				} else {
+					this.set('errorMessage', this.messageOnUnknown)
+				}
+				this.dispatchEvent(new CustomEvent('auth-error', {
+					detail: {
+						status: err && err.status,
+						message: this.errorMessage
+					}
+				}))
+			})
+
+			const defaultClient = getDefaultClient()
+
+			defaultClient.onEnterState(ClientStates.AUTHENTICATION_ERROR, updateMessagesOnError)
+			console.log('------> ',(defaultClient as any)._state)
+			const userCredentials: UserCredentials = new UserCredentials(this.username, this.password)
+			await defaultClient.updateCredentials(userCredentials)
+			await defaultClient.start()
+		}
+	}
+
+	private validateForm(): void {
+		this.$.username.validate()
+		this.$.password.validate()
+		if (!this.username.trim().length) return this.$.username.focus()
```

Comment:
always add curly braces to if statements, did you run and pass all the tslint rules ?